.github/workflows/tests.yml

```diff
@@ -110,27 +114,33 @@ jobs:
       S3_BACKEND_KEY_NAME: tf-state-${{ github.sha }}
     steps:
       - name: checkout code
+        if: ${{ env.AWS_ACCESS_KEY_ID }}
         uses: actions/checkout@v2.1.0
       - name: setup node
+        if: ${{ env.AWS_ACCESS_KEY_ID }}
         uses: actions/setup-node@v2
         with:
           node-version: '14'
       - uses: hashicorp/setup-terraform@v1
+        if: ${{ env.AWS_ACCESS_KEY_ID }}
         with:
           terraform_version: 0.12.25
           continue-on-error: false
       - name: rewrite tf state key
+        if: ${{ env.AWS_ACCESS_KEY_ID }}
         id: tf-state-key
         working-directory: ./infrastructure
         run: sed -i 's/BRANCH_ACTOR_KEY/${{ env.S3_BACKEND_KEY_NAME }}/g' infra.tf
 
       - name: Terraform Init
+        if: ${{ env.AWS_ACCESS_KEY_ID }}
         id: init
         working-directory: ./infrastructure
         run: terraform init
         continue-on-error: false
 
       - name: Terraform apply
+        if: ${{ env.AWS_ACCESS_KEY_ID }}
         id: apply
         working-directory: ./infrastructure
         run: terraform destroy -var="ec2_pwd=$EC2_SECRET" -var="branch_name=${{ github.sha }}" -auto-approve
```

Comment:
given the fact that the "runs-on" in the build phase has changed, the branch_name should have ${{ needs.pop-runner-instance.outputs.runner }} as a value;

the name of this variable is to be changed in the next PR

Reply:
```suggestion
        run: terraform destroy -var="ec2_pwd=$EC2_SECRET" -var="branch_name=${{ needs.pop-runner-instance.outputs.runner }}" -auto-approve
```